Purchases

jago425 opened the pull request:
Purchases resource is complete for Create, Index and Show - save for the Stripe Integration which we'll add in with another branch

jago425 committed:
Created Purchases

-added schema to model
-created purchases controller file
-created purchases scripts directory and wrote curl scripts for
create, index and show
-added purchases to routes for create, index and show only

Next Steps:
-write create action in the purchases controller

Rachel Jagodowski pair programming with Hilary Brown

jago425 committed:
Update user virtual to convert count string to number

Explicitly convert string item count to number before addition

jago425 committed:
Finish purchase create

Remove stripe user ID from purchase model
Add boiler plate code for all purchase methods
Complete purchase create method
Update purchase create curl script

Pair programming: Rachel Jagodowski Hilary Brown Jeff Keilman

jago425 committed:
Added CRUD for Index and Show

-curl scripts passed
-removed set model for update and destroy

Pair Programming: Rachel Jagodowski and Jeff Keilman

BrianLM commented on app/controllers/purchases.js:
```diff
@@ -0,0 +1,60 @@
+  create
+}, { before: [
+  { method: setUser, only: ['index', 'show', 'create'] },
+  { method: authenticate, except: ['index', 'show'] },
```
Remove except, keep method: authenticate

BrianLM commented on app/controllers/purchases.js:
```diff
@@ -0,0 +1,60 @@
+const setModel = require('./concerns/set-mongoose-model')
+
+const index = (req, res, next) => {
+  Purchase.find()
```
Purchase.find.(_owner: req.user._id)

jago425 committed:
Corrections to Purchases Index after code review

-added find method to assign owner to index
-removed exceptions from authentication in the before methods
-retested using the index curl script

Triple Threat Programming: Rachel Jagodowski, Brian DiStefano and Jeff
Keilman
